bugsnag-android-core/src/main/java/com/bugsnag/android/Event.kt

```diff
@@ -0,0 +1,174 @@
+package com.bugsnag.android
+
+import java.io.IOException
+import java.util.HashMap
+
+/**
+ * Information and associated diagnostics relating to a handled or unhandled
+ * Exception.
+ *
+ * This object is made available in OnError callbacks, so you can
+ * inspect and modify it before it is delivered to Bugsnag.
+ *
+ * @see OnError
+ */
+class Event @JvmOverloads internal constructor(
+    val originalError: Throwable? = null,
+    internal val config: ImmutableConfig,
+    private var handledState: HandledState,
+    internal val metadata: Metadata = Metadata(),
+    threadState: ThreadState? = if (config.sendThreads) {
+        ThreadState(config, originalError)
+    } else null
+) : JsonStream.Streamable, MetadataAware {
+
+    var session: Session? = null
+
+    /**
+     * Set the Severity of this Event.
+     *
+     * By default, unhandled exceptions will be Severity.ERROR and handled
+     * exceptions sent with bugsnag.notify will be Severity.WARNING.
+     * @see Severity
+     */
+    var severity: Severity
+        get() = handledState.currentSeverity
+        set(value) {
+            handledState.currentSeverity = value
+        }
+
+    var apiKey: String = config.apiKey
+
+    var app: MutableMap<String, Any?> = HashMap()
+    var device: MutableMap<String, Any?> = HashMap()
+
+    val unhandled = handledState.isUnhandled
```

Comment:
The spec currently states that this should be a readonly property named `unhandled`. Currently the property can be accessed in the following way via Kotlin and Java:

```kotlin
val: Boolean b = event.unhandled // Kotlin
```

```java
Boolean b = event.getUnhandled(); // Java
```

Reply:
The property name is fine but I would expect the Java syntax to be `isUnhandled()`

Reply:
As far as I can tell this would require us to name the Kotlin property `isUnhandled` in order to generate an `isUnhandled()` accessor: https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#properties

So this would be a question of whether we want to prioritise consistency with the spec for Kotlin or for Java.

Reply:
It looks to me like calling a boolean field `isX` is conventional in Kotlin and that would result in a Java API that we would expect. I don't think this means that we're violating the spec for Kotlin as `unhandled` is still the name of the property but expressed in a Kotlin way.

Reply:
Sounds good. We can use that rationale on any other public boolean properties throughout these updates.